dependencies/gtest/meson.build

```diff
@@ -0,0 +1,17 @@
+# gtest Dependency File
+#========================================================================
+
+
+# check if gtest is already installed
+#---------------------------------------------------
+gtest_dep = dependency(
+    'gtest',
+    required : false)
+
+
+# If not, include the submodule
+#---------------------------------------------------
+if (not gtest_dep.found())
+    gtest_proj = cmake.subproject('gtest')
```

Comment:
Why cmake is used here? As far as I remember, it is used by Meson as a last resort.

Also, as a typical nixpkgs developer, we are a bit afraid of expanding the closure of a project.

Reply:
it's just a direct port of the existing steps taken in the current buildsystem; it looks for the dependency, and if not, falls back on a built in version.

Reply:
OK then!

Reply:
i'd imagine it could be removed, but that wouldn't be my call to make

Reply:
Please explain me: if there is not a system-installed `gtest`, then it tries to build/use a locally maintained one?

Reply:
correct :)